client/analysisCtrl.ts

```diff
@@ -1048,10 +1179,12 @@ export default class AnalysisController {
             let position = (this.turnColor === this.mycolor) ? "bottom": "top";
             if (this.flip) position = (position === "top") ? "bottom" : "top";
             if (position === "top") {
-                this.pockets[0][role]++;
+                const pr = this.pockets[0][role];
+                if ( pr != undefined ) this.pockets[0][role] = pr + 1;

```

Comment:
i think i fixed all places, including irrelevant old code. also did the same for ==. 

I think i was somewhat careful so was not completely blindly replaced so hopefully no new bugs because of that, but  not bad to review maybe